lib/active_record/comments/execute_with_comments.rb

```diff
@@ -4,12 +4,19 @@ module ExecuteWithComments
       class << self
         def included(base)
           base.class_eval do
+            # ActiveRecord 7.1
+            if base.method_defined?(:internal_exec_query)
+              alias_method :exec_query_without_comment, :internal_exec_query
+              def internal_exec_query(query, *args, **kwargs, &block)
+                query = ActiveRecord::Comments.with_comment_sql(query)
+                exec_query_without_comment(query, *args, **kwargs, &block)
+              end
             # ActiveRecord 3.2 vs sqlite, maybe others ...
-            if base.method_defined?(:exec_query)
+            elsif base.method_defined?(:exec_query)
               alias_method :exec_query_without_comment, :exec_query
-              def exec_query(query, *args, prepare: false, &block)
+              def exec_query(query, *args, **kwargs, &block)
                 query = ActiveRecord::Comments.with_comment_sql(query)
-                exec_query_without_comment(query, *args, prepare: prepare, &block)
+                exec_query_without_comment(query, *args, **kwargs, &block)
               end
 
             # 99% case
```

Comment:
no longer the 99% right ?

Reply:
Nope, I went ahead and removed that comment.